src/Service/Factory/MailServiceAbstractFactory.php

```diff
@@ -85,6 +85,7 @@ public function __invoke(ContainerInterface $container, $requestedName, ?array $
         $specificServiceName = explode('.', $requestedName)[2] ?? null;
         $mailOptions = $container->get('config')['acmailer_options'] ?? [];
         $specificMailServiceOptions = $mailOptions['mail_services'][$specificServiceName] ?? null;
+        $throwOnCancel = $mailOptions['mail_services']['throw_on_cancel'] ?? false;
```

Comment:
As per my first comment, this should be

```suggestion
        $throwOnCancel = specificMailServiceOptions['throw_on_cancel'] ?? false;
```